site/src/sphinx/client-circuit-breaker.rst

```diff
@@ -111,40 +111,81 @@ is raised or the status is ``5xx``, succeeds when the status is ``2xx`` and igno
 
 .. code-block:: java
 
+    import com.linecorp.armeria.client.ClientRequestContext;
+    import com.linecorp.armeria.client.UnprocessedRequestException;
+    import com.linecorp.armeria.common.HttpStatus;
     import com.linecorp.armeria.common.HttpStatusClass;
 
     final CircuitBreakerStrategy<HttpResponse> myStrategy = new CircuitBreakerStrategy<HttpResponse>() {
 
         @Override
-        public CompletionStage<Boolean> shouldReportAsSuccess(HttpResponse response) {
-            return response.aggregate().handle((res, cause) -> {
-                if (cause != null) { // A failure if an Exception is raised.
-                    return false;
+        public CompletionStage<Boolean> shouldReportAsSuccess(ClientRequestContext ctx,
+                                                              @Nullable HttpResponse response,
+                                                              @Nullable Throwable cause) {
+            if (cause != null && !(cause instanceof UnprocessedRequestException)) {
+                // A failure if an Exception is raised.
+                return CompletableFuture.completedFuture(false);
+            }
+
+            final HttpStatus status = ctx.log().responseHeaders().status();
+            if (status != null) {
+                // A failure if the response is 5xx.
+                if (status.codeClass() == HttpStatusClass.SERVER_ERROR) {
+                    return CompletableFuture.completedFuture(false);
                 }
 
-                final HttpStatus status = res.status();
-                if (status != null) {
-                    // A failure if the response is 5xx.
-                    if (status.codeClass() == HttpStatusClass.SERVER_ERROR) {
-                        return false;
-                    }
-
-                    // A success if the response is 2xx.
-                    if (status.codeClass() == HttpStatusClass.SUCCESS) {
-                        return true;
-                    }
+                // A success if the response is 2xx.
+                if (status.codeClass() == HttpStatusClass.SUCCESS) {
+                    return CompletableFuture.completedFuture(true);
                 }
+            }
 
-                // Neither a success nor a failure. Do not take this response into account.
-                return null;
-            });
+            // Neither a success nor a failure. Do not take this response into account.
+            return CompletableFuture.completedFuture(null);
         }
     };
 
 If you want to treat a :api:`Response` as a success, return ``true``. Return ``false`` to treat as a failure.
 Note that :api:`CircuitBreakerStrategy` can return ``null`` as well. It won't be counted as a success nor
 a failure.
 
+You can use the actual ``HttpResponse`` to make a decision overriding
```

Comment:
How about:

If you need to determine if the request was successful by looking into the response content, you need to override ...